internal/postgresql/postgresql.go

```diff
@@ -838,10 +838,24 @@ func (p *Manager) SyncFromFollowed(followedConnParams ConnParams, replSlot strin
 	cmd.Env = append(os.Environ(), fmt.Sprintf("PGPASSFILE=%s", pgpass.Name()))
 	log.Debugw("execing cmd", "cmd", cmd)
 
-	// Pipe command's std[err|out] to parent.
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	if err := cmd.Run(); err != nil {
+	// Pipe pg_basebackup's stderr to our stderr.
+	// We do this indirectly so that pg_basebackup doesn't think it's connected to a tty.
+	// This ensures that it doesn't print any bare line feeds, which could corrupt other
+	// logs.
+	// pg_basebackup uses stderr for diagnostic messages and stdout for streaming the backup
+	// itself (in some modes; we don't use this). As a result we only need to deal with
+	// stderr.
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("error: %v", err)
+	}
+
+	cmd.Start()
+	if _, err := io.Copy(os.Stderr, stderr); err != nil {
+		return fmt.Errorf("error: %v", err)
```

Comment:
Returning an error here (also it's vary hard that this will happen) will keep a zombie process since `cmd.Wait()` will not be called.

I'll prefer to execute the io.Copy inside a go routine so the program function can go ahead and block on `cmd.Wait()`. The goroutine will ignore possible io.Copy errors since they're not really important.